content/docs/projects/csi-driver.md

```diff
@@ -1,42 +1,30 @@
 ---
 title: csi-driver
-description: ''
+description: 'Mounting cert-manager certificates without secrets'
 ---
 
-csi-driver is a Container Storage Interface (CSI) driver plugin for Kubernetes
-to work along cert-manager. The goal for this plugin is to seamlessly request
-and mount certificate key pairs to pods. This is useful for facilitating mTLS,
-or otherwise securing connections of pods with guaranteed present certificates
-whilst having all of the features that cert-manager provides.
-
-## Why a CSI Driver?
-
-- Ensure private keys never leave the node and are never sent over the network.
-  All private keys are stored locally on the node.
-- Unique key and certificate per application replica with a grantee to be
-  present on application run time.
-- Reduce resource management overhead by defining certificate request spec
-  in-line of the Kubernetes Pod template.
-- Automatic renewal of certificates based on expiry of each individual
-  certificate.
-- Keys and certificates are destroyed during application termination.
-- Scope for extending plugin behavior with visibility on each replica's
-  certificate request and termination.
-
-## Requirements and Installation
-
-This CSI driver plugin makes use of the 'CSI inline volume' feature - Alpha as
-of `v1.15` and beta in `v1.16`. Kubernetes versions `v1.16` and higher require
-no extra configuration however `v1.15` requires the following feature gate set:
-```
---feature-gates=CSIInlineVolume=true
-```
+csi-driver is a [Container Storage Interface (CSI)](https://kubernetes-csi.github.io/docs/) driver plugin for Kubernetes
+which works alongside cert-manager.
+
+Pods which mount the cert-manager csi-driver will request certificates from cert-manager
+without needing a `Certificate` resource to be created. These certificates will be mounted
+directly into the pod, with no intermediate Secret being created.
+
+## Why use csi-driver?
+
+- Ensure private keys never leave the node and are never sent over the network. All private keys are stored locally on the node
+- Unique key and certificate per application replica.
+- Fewer `Certificate` resources means writing less YAML
+- Keys and certificates are destroyed when an application terminates
+- No `Secret` resources needed for storing the certificate means less RBAC
+
```

Comment:
Could we also talk about mTLS here? This example was removed from the intro.

Reply:
We could, but I'm not really sure why we call out mTLS at all. I can do mTLS just fine without csi-driver just like I can get a server cert just fine without csi-driver, right?

Is mentioning mTLS just to drop the keyword "mTLS"? Is csi-driver helping in some way I'm missing here?

Reply:
Maybe we can state that this is a perfect solution for "ephemeral certificates" (eg. mTLS certificates).

I think it is important that we also clearly state that this is not a solution for long-lived certificates that should survive node restarts etc. (because each new pod instance will get a new cert).

Reply:
Good shout, yeah, I'll try to capture that!